Modules/Core/Common/include/itkMacro.h

```diff
@@ -1187,9 +1190,10 @@ compilers.
 
 /** Create members "name"On() and "name"Off() (e.g., DebugOn() DebugOff()).
  * Set method must be defined to use this macro. */
-#define itkBooleanMacro(name)                        \
-  virtual void name##On() { this->Set##name(true); } \
-  virtual void name##Off() { this->Set##name(false); }
+#define itkBooleanMacro(name)                          \
+  virtual void name##On() { this->Set##name(true); }   \
+  virtual void name##Off() { this->Set##name(false); } \
+  ITK_MACROEND_NOOP_STATEMENT
```

Comment:
Could this cause compiler errors for users who called `itkBooleanMacro(name)` _without_ semi-colon?

No problem to me, just wondering. 🤷

Reply:
@hjmjohnson Why did you mark this conversation as resolved?

Reply:
@N-Dekker I responded, but what I typed must not have been committed. Sorry.  This is partially related to fixing the -Wextra-semi (see #4724) warnings.   Some IDE's (clion) produce a lot of warnings about extra semi-colons, and clang-format requires a lot of additional work to avoid extra semi-colons.  
 
Yes, this could cause compiler errors this is an extension of the work started in 2020.    By consistently making macro uses end with a ; when used like a statement, both clang-format and the -Wextra-semi warnings are addressed, and the code looks consistent.

commit 72a45394b2905314f5cad18257cbf7028dae341a
Author: Hans J. Johnson <hans-johnson@uiowa.edu>
Date:   2020-03-18 10:05:19 -0500

    COMP: Fix extra semi-colon compiler warning
    
    ITK demands that macros end with a semi-colon so when
    the macro is expanded to nothing, there is an extra
    ; that is complained about.
    
    https://stackoverflow.com/questions/28777660/practical-differences-between-do-while-0-and-void0-in-macr/28777865
    https://stackoverflow.com/questions/1306611/how-do-i-implement-no-op-macro-or-template-in-c
    
    Define new ITK_STATEMENT_NOOP for consistent re-use in functions.
    
    pragma-push/pop operations can not have ';' at the end, so protect with
    "// clang-format [off|on]" syntax to ensure consistent formatting
    across different configuration options.

Reply:
Thanks for your elaborate reply, Hans! 👍  I'll mark this conversation resolved 😃